lib/sles4sap/azure_cli.pm

```diff
@@ -1374,3 +1378,163 @@ sub az_resource_delete {
 
     assert_script_run(join(' ', @az_command), timeout => $args{timeout});
 }
+
+=head2 az_storage_blob_upload
+
+    az_storage_blob_upload(
+        container_name=>'somecontainer',
+        storage_account_name=>'storageaccount',
+        file=>'somefilename' [, timeout=>42]);
+
+Uploads file to a storage container.
+
+=over 4
+
+=item B<container_name> Existing storage container name.
+
+=item B<storage_account_name> Storage account name.
+
+=item B<file> File to upload.
+
+=item B<timeout> Timeout for az command.
+
+=back
+=cut
+
+sub az_storage_blob_upload {
+    my (%args) = @_;
+    foreach ('container_name', 'storage_account_name', 'file') {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    $args{timeout} //= '90';
+
+    my $az_cmd = join(' ',
+        'az storage blob upload',
+        '--only-show-errors',
+        "--container-name $args{container_name}",
+        "--account-name $args{storage_account_name}",
+        "--file $args{file}"
+    );
+    assert_script_run(join(' ', $az_cmd), timeout => $args{timeout});
+}
+
+=head2 az_storage_blob_lease_acquire
+
+    az_storage_blob_lease_acquire(
+        container_name=>'somecontainer',
+        storage_account_name=>'storageaccount',
+        blob_name => 'somefilename' [, lease_duration=>'42']
+    );
+
+Acquire a lease for a storage blob.
+
+=over 4
+
+=item B<container_name> Existing storage container name.
+
+=item B<storage_account_name> Storage account name.
+
+=item B<blob_name> Blob name to acquire lease for.
+
+=item B<lease_duration> Lease duration between 15-60s. Default: infinite
+
+=back
+=cut
+
+sub az_storage_blob_lease_acquire {
+    my (%args) = @_;
+    foreach ('container_name', 'storage_account_name', 'blob_name') {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    $args{lease_duration} //= '-1';    # -1 = infinite lease
+
+    my $az_cmd = join(' ',
+        'az storage blob lease acquire',
+        '--only-show-errors',
+        "--container-name $args{container_name}",
+        "--account-name $args{storage_account_name}",
+        "--blob-name $args{blob_name}",
+        "--lease-duration $args{lease_duration}",
+        '--output tsv'    # Json output won't work here.
+    );
+
+    my $lease_id = script_output($az_cmd, proceed_on_failure => 1);
+    return ($lease_id) if qesap_az_validate_uuid_pattern($lease_id);    # Return only valid output.
```

Comment:
could you please move the documentatio about the return code in the function documentation. How it is expected to works in case of error, or better to say how the caller should handle the "empty return" in case of error?

Reply:
I see, like this

```
$lease_id = az_storage_blob_lease_acquire(%arguments);

    # Return unless there file lease was successful
    return unless $lease_id;
```

Reply:
The purpose of this function is to request exclusive file rights by calling `az storage blob lease acquire`. If call is successful, it will return UUID which you use afterwards for accessing the file. 

So if the function returns UUID, the file is yours. If it returns nothing, it means you won't get the rights and is up to the caller to decide how to deal with this situation - it can make the test fail completely, or retry in a loop. 

I just updated the description to:
 ```
Acquire a lease for a storage blob. Function returns UUID which is then required to modify the file and gives the
UUID owner exclusive rights.
Optionally B<lease_duration> can be defined to limit this file lock up to 60s instead of infinity.

```